ui/src/app/views/workspace-share/component.ts

```diff
@@ -0,0 +1,86 @@
+import {Location} from '@angular/common';
+import {Component, OnInit} from '@angular/core';
+import {Router, ActivatedRoute} from '@angular/router';
+
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+import {isBlank} from 'app/utils';
+
+import {DataAccessLevel} from 'generated';
+import {UserRole} from 'generated';
+import {UserRoleList} from 'generated';
+import {Workspace} from 'generated';
+import {WorkspaceAccessLevel} from 'generated';
+import {WorkspacesService} from 'generated';
+
+@Component({
+  styleUrls: ['./component.css'],
+  templateUrl: './component.html',
+})
+export class WorkspaceShareComponent implements OnInit {
+  workspace: Workspace;
+  loadingWorkspace = true;
+  toShare = '';
+  selectedPermission = 'Select Permission';
+  accessLevel: WorkspaceAccessLevel;
+  constructor(
+      private errorHandlingService: ErrorHandlingService,
+      private locationService: Location,
+      private router: Router,
+      private route: ActivatedRoute,
+      private workspacesService: WorkspacesService,
+  ) {}
+
+  ngOnInit(): void {
+    this.errorHandlingService.retryApi(
+        this.workspacesService.getWorkspace(
+            this.route.snapshot.params['ns'],
+            this.route.snapshot.params['wsid']))
+      .subscribe((workspace) => {
+        this.loadingWorkspace = false;
+        this.workspace = workspace;
+      }
+    );
+  }
+
+  navigateBack(): void {
+    this.locationService.back();
+  }
+
+  setAccess(dropdownSelected: string): void {
+    this.selectedPermission = dropdownSelected;
+    if (dropdownSelected === 'Owner') {
+      this.accessLevel = WorkspaceAccessLevel.Owner;
+    } else if (dropdownSelected === 'Writer') {
+      this.accessLevel = WorkspaceAccessLevel.Writer;
+    } else {
+      this.accessLevel = WorkspaceAccessLevel.Reader;
+    }
+  }
+
+  addCollaborator(): void {
+    this.workspace.userRoles.push({user: this.toShare, role: this.accessLevel});
+    const userRoleList: UserRoleList = {items: this.workspace.userRoles};
+    this.workspacesService.shareWorkspace(this.workspace.namespace,
+        this.workspace.id, userRoleList).subscribe(
+      () => {
+      }
+    );
+  }
+
+  removeCollaborator(user: UserRole): void {
```

Comment:
Error message if you try to remove yourself and you're the last owner?

Reply:
Will enforce that API side if I can.

Reply:
Actually, I talked to Mohs, and Firecloud already does this. So I think we can enforce this when we add Firecloud sharing to our API, so that if FC errors we don't continue by sharing in our DB. (And that allows us to avoid repeating logic and code.)